[Manual Taxes M#1] Tax rates education dialogDisplay tax lines and "tax based on" setting in order creation

samiuelson opened the pull request:
<!-- Remember about a good descriptive title. -->
#### This PR adds displaying of tax lines and "tax based on" setting in order creation screen.

Closes: #9625
<!-- Id number of the GitHub issue this PR addresses. -->

⚠️ Do not merge before corresponding PR in FluxC is merged and FluxC version is updated in this PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2812

Note - there are the following follow-up PRs:
- https://github.com/woocommerce/woocommerce-android/pull/9651
- https://github.com/woocommerce/woocommerce-android/pull/9647

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR displays active tax rate information - tax lines from the Order model with the "Tax based on" setting from `wp-admin > WooCommerce > Settings > Tax`.

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->
You'll need to test with the following FluxC branch:  https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2812 or review and merge FluxC PR first.

Verify you have tax rates defined under `wp-admin > WooCommerce > Settings > Tax > Standard Rates`
Verify your tax-based-on setting under `wp-admin > WooCommerce > Settings > Tax`
1. Create new order
2. Add product
3. Observe tax lines being displayed in the payments section (tax rate name + "tax based on" value)
Try the above with the modified settings.

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->
<img width=300 src=https://github.com/woocommerce/woocommerce-android/assets/4527432/ef79deef-ee9d-432e-986d-dda99e3b1d9d/>

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.
I'll update notes in the next PR: tax rates education dialog.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

samiuelson committed:
Display tax lines and "tax based on" setting in order creation

samiuelson committed:
Add unit tests

samiuelson committed:
Add unit tests

samiuelson committed:
Fix unit tests

samiuelson committed:
Clean up code

samiuelson committed:
Clean up code

samiuelson committed:
Fix detekt issues

rossanafmenezes commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/TaxLines.kt:
```diff
@@ -0,0 +1,53 @@
+fun TaxLines(taxLines: List<Order.TaxLine>?, taxSettingLabel: String?) {
+    Column {
+        taxLines?.forEach { taxLine ->
+            Row(
```
🤔  Just wondering if it would be ok to add the tax name and setting label in two different lines because when you have a big tax name it looks a bit weird:
![Screenshot 2023-08-23 at 1 57 21 PM](https://github.com/woocommerce/woocommerce-android/assets/30724184/3311bdf4-481c-4788-951a-be9eb0941685)

So I was wondering if having it in separate line (but keeping the font style and the minimal padding between the two lines it would still give the idea that they are related but give the tax name more space.

samiuelson replied:
Good point! I've addressed this issue in a separate PR: https://github.com/woocommerce/woocommerce-android/pull/9651

rossanafmenezes reviewed (approved):
LGTM other than the CI tests

samiuelson commented:
> LGTM other than what I mentioned and the CI tests

Thanks for the review. The CI checks should pass once the underlying FluxC PR is merged into trunk.

samiuelson committed:
Update code to match new data model from FluxC

samiuelson committed:
Update imports after fluxc changes

samiuelson committed:
Update FluxC version

samiuelson committed:
Merge branch 'trunk' into 9624/tax-lines-in-order-creation

samiuelson committed:
Fix detekt issues

samiuelson committed:
Fix unit tests

samiuelson committed:
Update unit tests

samiuelson committed:
Update e2e test

samiuelson committed:
Update OrderCreateEditRepositoryTest

pachlava commented:
@samiuelson 👋 🙂  

Just leaving a note that having `failOnUnmatchedWireMockRequests` set to `true` is a conscious decision for Android - see discussion https://github.com/woocommerce/woocommerce-android/pull/9597/files#r1295819733. I added a missing mock for the request, introduced by this PR, and reverted the `failOnUnmatchedWireMockRequests` value back to default with #9679.

samiuelson commented:
> Just leaving a note that having `failOnUnmatchedWireMockRequests` set to `true` is a conscious decision for Android - see discussion https://github.com/woocommerce/woocommerce-android/pull/9597/files#r1295819733. I added a missing mock for the request, introduced by this PR, and reverted the `failOnUnmatchedWireMockRequests` value back to default with #9679.

Thanks for the explanation and for updating the mocks, @pachlava. I wasn't aware of this.
